src/nnet3bin/nnet3-copy-egs.cc

```diff
@@ -301,7 +340,18 @@ int main(int argc, char *argv[]) {
                 "feature left-context that we output.");
     po.Register("right-context", &right_context, "Can be used to truncate the "
                 "feature right-context that we output.");
-
+    po.Register("weights", &weight_str,
+                "Rspecifier of a table indexed by the key of input examples"
+                "and value of scalar weights."
+                "If provided, the supervision output in eg is scaled using weight."
+                " Scaling output supervision is the same as scaling the "
+                "derivative during training for linear objective."
+                "The default is one, which means we are not applying per-example weights.");
+    po.Register("outputs", &output_str,
```

Comment:
The names weren't ideal, but I think it's OK to leave them.
However, the usage messages are extremely unclear.  You should start out by saying that they are rspecifiers indexed by the same string as is used to index the utterance-id.

The variable names should be e.g. output_rxfilename and input_rxfilename.